### PR TITLE
 * StringOutput set proper ContentType

### LINKED ISSUE #3692: ASP.NET MVC returns Content-Type text/plain even if application/xml was requested in case of return type is string
This is my controller:

```c#
[Route("")]
public class ValuesController : Controller
{
    //This always returns text/plain
    [HttpPost("/text")]
    [Produces("application/xml", "application/json")]
    public string GetText() => "Some text";

    [HttpPost("/textarray")]
    [Produces("application/xml", "application/json")]
    public IEnumerable<string> GetTextarray() => new[] { "Some text", "Some text2" };
}
```

GetText() always returns text/plain even if I specify "Accept: application/xml" in request header. This happens only in case of return type "string".  I expect get xml instead of plain text. See sample requests below:

Request 1:
GET /text HTTP/1.1
Host: localhost:5004
Accept: application/xml

Response 1:
HTTP/1.1 200 OK
Date: Fri, 04 Dec 2015 20:13:00 GMT
Content-Type: text/plain; charset=utf-8
Server: Kestrel

Some text

Request 2:
GET /textarray HTTP/1.1
Host: localhost:5004
Accept: application/xml

Response 2:
HTTP/1.1 200 OK
Date: Fri, 04 Dec 2015 20:13:04 GMT
Content-Type: application/json; charset=utf-8
Server: Kestrel

```xml
<ArrayOfString xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema"><string>Some text</string><string>Some text2</string></ArrayOfString>
```

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes #3692

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Core/Formatters/StringOutputFormatter.cs [hunk 1]
```diff
@@ -34,8 +34,12 @@ public override bool CanWriteResult(OutputFormatterCanWriteContext context)
             // always return it as a text/plain format.
             if (context.ObjectType == typeof(string) || context.Object is string)
             {
-                context.ContentType = new StringSegment(SupportedMediaTypes[0]);
-                return true;
+                if (context.ContentType == null)
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/Formatters/StringOutputFormatter.cs]: `StringSegment`s can't be null.

**Comment by rynowak** [comment]
Tests?

**Comment by rynowak** [comment]
:watch:

**Comment by ryanbrandenburg** [comment]
:up: :date:

File: src/Microsoft.AspNet.Mvc.Core/Formatters/StringOutputFormatter.cs [hunk 2]
```diff
@@ -34,7 +33,11 @@ public override bool CanWriteResult(OutputFormatterCanWriteContext context)
             // always return it as a text/plain format.
             if (context.ObjectType == typeof(string) || context.Object is string)
             {
-                context.ContentType = new StringSegment(SupportedMediaTypes[0]);
+                if (!context.ContentType.HasValue)
+                {
+                    context.ContentType = new StringSegment(SupportedMediaTypes[0]);
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/Formatters/StringOutputFormatter.cs]: Ideally we'd also include an charset here. Look at what the other formatters do

File: test/Microsoft.AspNet.Mvc.Core.Test/Formatters/StringOutputFormatterTests.cs [hunk 3]
```diff
@@ -35,9 +35,7 @@ public static IEnumerable<object[]> OutputFormatterContextValues
             bool expectedCanWriteResult)
         {
             // Arrange
-            var expectedContentType = expectedCanWriteResult ?
-                new StringSegment("text/plain") :
-                new StringSegment("application/json");
+            var expectedContentType = new StringSegment("application/json");
 
             var formatter = new StringOutputFormatter();
             var type = useDeclaredTypeAsString ? typeof(string) : typeof(object);
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/Formatters/StringOutputFormatterTests.cs]: Where's the unit test for the new behavior?

**Comment by rynowak** [comment]
:watch:

**Comment by ryanbrandenburg** [comment]
:up: :date: Added unit tests and set charset on default ContentType.

File: src/Microsoft.AspNet.Mvc.Core/Formatters/StringOutputFormatter.cs [hunk 4]
```diff
@@ -20,7 +19,7 @@ public StringOutputFormatter()
         {
             SupportedEncodings.Add(Encoding.UTF8);
             SupportedEncodings.Add(Encoding.Unicode);
-            SupportedMediaTypes.Add("text/plain");
+            SupportedMediaTypes.Add("text/plain;charset=utf-8");
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/Formatters/StringOutputFormatter.cs]: This isn't right. Look at what the other formatters do.

**Comment by ryanbrandenburg** [comment]
:up: :date:

**Comment by rynowak** [comment]
:shipit:

**Pull request merged**